lib/martian_robot.rb

```diff
@@ -1,25 +1,30 @@
+require_relative 'movements'
+require_relative 'mars'
+
 class MartianRobot
 
   attr_reader :coordinates, :orientation
 
-  def initialize(coordinates, orientation)
+  def initialize(coordinates, orientation, size)
     @coordinates = coordinates
     @orientation = orientation
     @movements = Movements.new
+    @planet = Mars.new(size)
   end
 
   def report_final_position(instruction)
     final_position = starting_position
     instruction.split('').each do |single_instruction|
-      if single_instruction == 'F'
+      case single_instruction
+      when'F'
         @movements.move_forward(@orientation, final_position)
-      elsif single_instruction == 'R'
+      when 'R'
```

Comment:
The instruction could be embedded into a private method.

This will let you have a shorter method and easier to follow:

``` ruby
instructions.split('').each do |instruction|
  execture(instruction)
end
```

which could be write also this way:
``` ruby
instructions.split('').each { |instruction| execute(instruction) }
```